tests/west/project/test_project.py

```diff
@@ -427,6 +428,169 @@ def test_extension_command_execution(west_init_tmpdir):
+
+    add_commit(remote_zephyr, 'test added extension command',
+               files={'west.yml': textwrap.dedent('''\
+                      west:
+                        url: file://{west}
+                      manifest:
+                        defaults:
+                          remote: test-local
+
+                        remotes:
+                          - name: test-local
+                            url-base: file://{rr}
+
+                        projects:
+                          - name: Kconfiglib
+                            revision: zephyr
+                            path: subdir/Kconfiglib
+                            west-commands: scripts/west-commands.yml
+                          - name: net-tools
+                            west-commands: scripts/west-commands.yml
+                        self:
+                          path: zephyr
+                      '''.format(west=remote_west, rr=str(rr)))})
+
+    # Initialize the net-tools repository.
+    add_commit(remote_kconfiglib, 'add west commands',
+               files={'scripts/west-commands.yml': textwrap.dedent('''\
+                      west-commands:
+                        - file: scripts/test.py
+                          commands:
+                            - name: kconfigtest
+                              class: Test
+                      '''),
+                      'scripts/test.py': textwrap.dedent('''\
+                      from west.commands import WestCommand
+                      class Test(WestCommand):
+                          def __init__(self):
+                              super(Test, self).__init__(
+                                  'kconfigtest',
+                                  'Kconfig test application',
+                                  '')
+                          def do_add_parser(self, parser_adder):
+                              parser = parser_adder.add_parser(self.name)
+                              return parser
+                          def do_run(self, args, ignored):
+                              print('Testing kconfig test')
+                      '''),
+                      })
+    west_tmpdir = repos_tmpdir.join('west_installation')
+    cmd('init -m "{}" "{}"'.format(str(repos_tmpdir.join('repos', 'zephyr')),
+                                   str(west_tmpdir)))
+    west_tmpdir.chdir()
+    config.read_config()
+    cmd('update')
+
+    help_text = cmd('-h').splitlines()
+
+    cmd_help = []
+    for line in help_text:
+        if cmd_help or re.match(r'commands from project', line):
+            cmd_help.append(line)
+        else:
+            print(line)
+
+    expected = [
+        'commands from project at "subdir/Kconfiglib":',
+        '  kconfigtest',
+        '',
+        'commands from project at "net-tools":',
+        '  test',
+        '',
+        'Run "west <command> -h" for help on each command.',
+        '']
+    assert len(expected) == len(cmd_help), "expected:\n{}\nbut was:\n{}"\
```

Comment:
Pytest is really smart about runtime type inspecting the arguments of either side of the assert. It will provide very useful feedback if there is a mismatch. Check this output for example:

```
=============================================== FAILURES ===============================================
_______________________________________________ test_foo _______________________________________________

    def test_foo():
        expected = [1,2,3,4]
        actual = [1,2,'a',4]
>       assert expected == actual
E       AssertionError: assert [1, 2, 3, 4] == [1, 2, 'a', 4]
E         At index 2 diff: 3 != 'a'
E         Use -v to get the full diff
```

So I think you can just write `assert expected == actual`, making the test easier to read but not compromising debuggability.